packages/runtime/src/helpers/redirects.ts

```diff
@@ -143,8 +143,8 @@ const generateStaticIsrRewrites = ({
       return
     }
     // The default locale is served from the root, not the localised path
-    if (i18n?.defaultLocale && route.startsWith(`/${i18n.defaultLocale}/`)) {
-      route = route.slice(i18n.defaultLocale.length + 1)
+    if (i18n?.defaultLocale && route.startsWith(`/${i18n.defaultLocale}`)) {
```

Comment:
Wouldn't this cause false positives? e.g. if there's a site with `en` default locale, but also a route called `/entry`

Reply:
hi @ascorbic, it doesn't appear to - both this and the previous condition would match `/en/entry`

Reply:
Right, but would this match `/entry` and turn it into `try`?

Reply:
ah, i see what you mean - no it won't because with i18n the routes are always matched with the locale prefix, so for the route `/entry` it will be matched against `/en/entry` and `/fr/entry`, etc.

Reply:
Aah, gotcha. Unfortunately I just checked the manifest for the default demo in my ISR branch and it looks like appDir pages _don't_ have the prefix.

Reply:
great catch, i'd forgotten that appDir doesn't implement i18n - i've made the condition more specific, which is a good thing anyway [7cef5fe](https://github.com/netlify/next-runtime/pull/1867/commits/7cef5fefc1841fce80b61a79d734d24a900517de)